src/main/java/com/ably/kafka/connect/client/BatchSpec.java

```diff
@@ -0,0 +1,34 @@
+package com.ably.kafka.connect.client;
+
+import io.ably.lib.types.Message;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+public class BatchSpec {
```

Comment:
Minor: the `equals()` implementation here expects the class to be final, as sub-class instances will always return `false` because of the `getClass()` comparison. Personally I'd just make it `final` because it rarely makes sense to use inheritance on value types.

Reply:
Changed to final.